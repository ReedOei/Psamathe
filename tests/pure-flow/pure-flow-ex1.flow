;

true --> var b : ! bool
[ b ] --> var x : any list ! bool
true --> b
[ b ] --> x

