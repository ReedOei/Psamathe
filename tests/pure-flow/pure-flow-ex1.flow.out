<generatedTop>
  <k>
    .
  </k>
  <functions>
    .Map
  </functions>
  <freshCounter>
    6
  </freshCounter>
  <lookup>
    b |-> loc ( 0 )
    x |-> loc ( 2 )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( bool , .List )
    loc ( 1 ) |-> resource ( bool , .List )
    loc ( 2 ) |-> resource ( list ! bool , ListItem ( ListItem ( true ) )
    ListItem ( ListItem ( true ) ) )
    loc ( 3 ) |-> resource ( list ! bool , .List )
    loc ( 4 ) |-> resource ( bool , .List )
    loc ( 5 ) |-> resource ( list ! bool , .List )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
